From 6e36c4bb9fee4d1f474705938fad6f4d6fb9b32d Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Date: Fri, 8 Mar 2019 18:57:38 -0800
Subject: [PATCH 0892/1239] mmc: octeontx: octeontx2: move to new mmc driver

This migrates the configuration to the new eMMC driver.

Change-Id: I928c0add6277aa5449560a18dce0ff12d91f7e2d
Signed-off-by: aaronw <aaron.williams@cavium.com>
Signed-off-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/5427
---
 configs/octeontx2_95xx_defconfig | 11 ++++++++++-
 configs/octeontx2_96xx_defconfig |  7 +++++++
 configs/octeontx_81xx_defconfig  | 11 +++++++++--
 configs/octeontx_83xx_defconfig  | 11 +++++++++--
 include/configs/octeontx2_95xx.h |  5 +++++
 include/configs/octeontx2_96xx.h |  5 +++++
 include/configs/octeontx_81xx.h  |  5 +++++
 include/configs/octeontx_83xx.h  |  5 +++++
 8 files changed, 55 insertions(+), 5 deletions(-)

diff --git a/configs/octeontx2_95xx_defconfig b/configs/octeontx2_95xx_defconfig
index 4c02324803..bd15b63708 100644
--- a/configs/octeontx2_95xx_defconfig
+++ b/configs/octeontx2_95xx_defconfig
@@ -48,11 +48,20 @@ CONFIG_OF_BOARD=y
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_BLOCK_CACHE=y
+CONFIG_BLK=y
 CONFIG_DM_GPIO=y
+CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-# CONFIG_MMC is not set
+CONFIG_MMC=y
+CONFIG_MMC_HS200_SUPPORT=y
+CONFIG_MMC_HS400_SUPPORT=y
+CONFIG_MMC_WRITE=y
+CONFIG_MMC_SUPPORTS_TUNING=y
 CONFIG_DM_MMC=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
diff --git a/configs/octeontx2_96xx_defconfig b/configs/octeontx2_96xx_defconfig
index b01da23a73..53765c6c4f 100644
--- a/configs/octeontx2_96xx_defconfig
+++ b/configs/octeontx2_96xx_defconfig
@@ -24,6 +24,11 @@ CONFIG_CMD_MX_CYCLIC=y
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
+CONFIG_MMC_HS200_SUPPORT=y
+CONFIG_MMC_HS400_SUPPORT=y
+CONFIG_MMC_WRITE=y
+CONFIG_MMC_SUPPORTS_TUNING=y
+CONFIG_MMC=y
 CONFIG_CMD_PART=y
 CONFIG_CMD_PCI=y
 CONFIG_CMD_SF=y
@@ -53,8 +58,10 @@ CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_SCSI_AHCI=y
 CONFIG_AHCI_PCI=y
 CONFIG_BLOCK_CACHE=y
+CONFIG_BLK=y
 CONFIG_NVME=y
 CONFIG_DM_GPIO=y
+CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_I2C_SET_DEFAULT_BUS_NUM=y
 CONFIG_I2C_MUX=y
diff --git a/configs/octeontx_81xx_defconfig b/configs/octeontx_81xx_defconfig
index af3deeeed1..af941d2091 100644
--- a/configs/octeontx_81xx_defconfig
+++ b/configs/octeontx_81xx_defconfig
@@ -56,11 +56,18 @@ CONFIG_AHCI_PCI=y
 CONFIG_BLOCK_CACHE=y
 CONFIG_NVME=y
 CONFIG_DM_GPIO=y
+CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-# CONFIG_MMC is not set
+CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_MMC_OCTEONTX=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+# CONFIG_MMC_HS200_SUPPORT is not set
+# CONFIG_MMC_HS400_SUPPORT is not set
+CONFIG_MMC_WRITE=y
+# CONFIG_MMC_SUPPORTS_TUNING is not set
 CONFIG_MTD=y
 CONFIG_MTD_PARTITIONS=y
 CONFIG_CMD_MTDPARTS=y
diff --git a/configs/octeontx_83xx_defconfig b/configs/octeontx_83xx_defconfig
index 0c84c44a97..b5b47212c1 100644
--- a/configs/octeontx_83xx_defconfig
+++ b/configs/octeontx_83xx_defconfig
@@ -56,11 +56,18 @@ CONFIG_AHCI_PCI=y
 CONFIG_BLOCK_CACHE=y
 CONFIG_NVME=y
 CONFIG_DM_GPIO=y
+CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-# CONFIG_MMC is not set
+CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_MMC_OCTEONTX=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+# CONFIG_MMC_HS200_SUPPORT is not set
+# CONFIG_MMC_HS400_SUPPORT is not set
+CONFIG_MMC_WRITE=y
+# CONFIG_MMC_SUPPORTS_TUNING is not set
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
diff --git a/include/configs/octeontx2_95xx.h b/include/configs/octeontx2_95xx.h
index a55f50a0ea..87aa5d4d9a 100644
--- a/include/configs/octeontx2_95xx.h
+++ b/include/configs/octeontx2_95xx.h
@@ -96,4 +96,9 @@
 #undef CONFIG_SYS_PROMPT
 #define CONFIG_SYS_PROMPT		env_get("prompt")
 
+#define CONFIG_MMC_SUPPORTS_TUNING
+#define MMC_SUPPORTS_TUNING
+#define CONFIG_CMD_BKOPS_ENABLE
+#define CONFIG_SUPPORT_EMMC_RPMB
+
 #endif /* __OCTEONTX2_95XX_H__ */
diff --git a/include/configs/octeontx2_96xx.h b/include/configs/octeontx2_96xx.h
index acd890ab67..f713400ab5 100644
--- a/include/configs/octeontx2_96xx.h
+++ b/include/configs/octeontx2_96xx.h
@@ -107,4 +107,9 @@
 #undef CONFIG_SYS_PROMPT
 #define CONFIG_SYS_PROMPT		env_get("prompt")
 
+#define CONFIG_MMC_SUPPORTS_TUNING
+#define MMC_SUPPORTS_TUNING
+#define CONFIG_CMD_BKOPS_ENABLE
+#define CONFIG_SUPPORT_EMMC_RPMB
+
 #endif /* __OCTEONTX2_96XX_H__ */
diff --git a/include/configs/octeontx_81xx.h b/include/configs/octeontx_81xx.h
index 18bf0fa1a6..22543e9e0c 100644
--- a/include/configs/octeontx_81xx.h
+++ b/include/configs/octeontx_81xx.h
@@ -120,4 +120,9 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_CMD_BKOPS_ENABLE
 
+#define CONFIG_MMC_OCTEONTX
+#ifndef CONFIG_BLK
+# define CONFIG_BLK
+#endif
+
 #endif /* __OCTEONTX_81XX_H__ */
diff --git a/include/configs/octeontx_83xx.h b/include/configs/octeontx_83xx.h
index 85b8f6306a..691448abc8 100644
--- a/include/configs/octeontx_83xx.h
+++ b/include/configs/octeontx_83xx.h
@@ -119,4 +119,9 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_CMD_BKOPS_ENABLE
 
+#define CONFIG_MMC_OCTEONTX
+#ifndef CONFIG_BLK
+# define CONFIG_BLK
+#endif
+
 #endif /* __OCTEONTX_83XX_H__ */
-- 
2.29.0

